sqldelight-compiler/src/test/kotlin/app/cash/sqldelight/core/QueriesTypeTest.kt

```diff
@@ -150,6 +153,7 @@ class QueriesTypeTest {
       |    |FROM data
       |    |WHERE id = ?
       |    ""${'"'}.trimMargin(), 1) {
+      |      check(this is SqlPreparedStatement)
```

Comment:
we should special case `SqlPreparedStatement` and `SqlCursor` to not require this check, and that should clean up a lot of the test cases